core/unit_test/TestMDTeamParallelism.hpp

```diff
@@ -0,0 +1,1660 @@
+
+    Kokkos::parallel_for(
+        Kokkos::MDRangePolicy<ExecSpace, Kokkos::Rank<5>>(
+            {0, 0, 0, 0, 0}, {leagueSize, n0, n1, n2, n3}),
+        KOKKOS_LAMBDA(const int i, const int j, const int k, const int l,
+                      const int m) {
+          v(i, j, k, l, m) = fillFlattenedIndex(i, j, k, l, m);
+        });
+
+    int finalSum = 0;
+
+    Kokkos::parallel_reduce(
+        Kokkos::TeamPolicy<ExecSpace>(leagueSize, Kokkos::AUTO),
+        KOKKOS_LAMBDA(TeamType const& team, int& leagueSum) {
+          auto leagueRank = team.league_rank();
+          int teamSum     = 0;
+
+          Kokkos::parallel_reduce(
+              Kokkos::MDTeamThreadRange<Direction>(team, n0, n1, n2, n3),
+              [=](const int& i, const int& j, const int& k, const int& l,
+                  int& threadSum) { threadSum += v(leagueRank, i, j, k, l); },
+              teamSum);
+
+          Kokkos::single(Kokkos::PerTeam(team),
+                         [&leagueSum, teamSum]() { leagueSum += teamSum; });
+        },
+        finalSum);
+
+    int expectedSum = get_expected_sum(dims, 5, fillFlattenedIndex);
+
+    EXPECT_EQ(finalSum, expectedSum);
+  }
+
+  template <Kokkos::Iterate Direction = Kokkos::Iterate::Default>
+  static void test_parallel_reduce_for_6D_MDTeamThreadRange(
+      DimsType const& dims) {
+    using ViewType = typename Kokkos::View<DataType******, ExecSpace>;
+
+    int leagueSize = dims[0];
+    int n0         = dims[1];
+    int n1         = dims[2];
+    int n2         = dims[3];
+    int n3         = dims[4];
+    int n4         = dims[5];
+
+    ViewType v("v", leagueSize, n0, n1, n2, n3, n4);
+    FillFlattenedIndex fillFlattenedIndex(leagueSize, n0, n1, n2, n3, n4);
+
+    Kokkos::parallel_for(
+        Kokkos::MDRangePolicy<ExecSpace, Kokkos::Rank<6>>(
+            {0, 0, 0, 0, 0, 0}, {leagueSize, n0, n1, n2, n3, n4}),
+        KOKKOS_LAMBDA(const int i, const int j, const int k, const int l,
+                      const int m, const int n) {
+          v(i, j, k, l, m, n) = fillFlattenedIndex(i, j, k, l, m, n);
+        });
+
+    int finalSum = 0;
+
+    Kokkos::parallel_reduce(
+        Kokkos::TeamPolicy<ExecSpace>(leagueSize, Kokkos::AUTO),
+        KOKKOS_LAMBDA(TeamType const& team, int& leagueSum) {
+          auto leagueRank = team.league_rank();
+          int teamSum     = 0;
+
+          Kokkos::parallel_reduce(
+              Kokkos::MDTeamThreadRange<Direction>(team, n0, n1, n2, n3, n4),
+              [=](const int& i, const int& j, const int& k, const int& l,
+                  const int& m, int& threadSum) {
+                threadSum += v(leagueRank, i, j, k, l, m);
+              },
+              teamSum);
+
+          Kokkos::single(Kokkos::PerTeam(team),
+                         [&leagueSum, teamSum]() { leagueSum += teamSum; });
+        },
+        finalSum);
+
+    int expectedSum = get_expected_sum(dims, 6, fillFlattenedIndex);
+
+    EXPECT_EQ(finalSum, expectedSum);
+  }
```

Comment:
Any reason why you didn't do 7D and 8D?

Reply:
Initially, I thought we didn't include 7D and 8D to have MD team range policies to stay consistent with `MDRangePolicy`, which requires its rank to be less than 7. After some reviewing, I think the real reason was because this test was using `MDRangePolicy` to construct the array of expected results, which wouldn't work for 7D and 8D. I'll change how that array is constructed and add in 7D and 8D for the reduction tests.